packages/agreement-process/src/services/agreementService.ts

```diff
@@ -345,6 +345,39 @@ export function agreementServiceBuilder(
         })
       );
 
+      return agreementId;
+    },
+    async archiveAgreement(
+      agreementId: string,
+      authData: AuthData
+    ): Promise<Agreement["id"]> {
+      logger.info(`Archiving agreement ${agreementId}`);
+
+      const agreement = await agreementQuery.getAgreementById(agreementId);
+      assertAgreementExist(agreementId, agreement);
+      assertRequesterIsConsumer(agreement.data, authData);
+
+      const updateSeed = {
+        ...agreement.data,
+        state: agreementState.archived,
+        stamps: {
+          ...agreement.data.stamps,
+          archiving: createStamp(authData),
+        },
+      };
+
+      const updatedAgreement = {
+        ...agreement.data,
+        ...updateSeed,
+      };
+
+      await repository.createEvent(
+        toCreateEventAgreementUpdated(
+          updatedAgreement,
+          agreement.metadata.version
+        )
+      );
+
```

Comment:
As we did with the `update`, should we track on Jira the implementation of the two "side effects" that this action shall have on eService Descriptors and on Purposes?

<img width="922" alt="image" src="https://github.com/pagopa/interop-be-monorepo/assets/6418684/49267308-27c2-4063-a28b-ede3d53b17cc">

Reply:
Maybe the descriptors archiviation is already tracked [here](https://pagopa.atlassian.net/browse/IMN-162), so we should track only the purposes part

Reply:
sure good suggestion 👍🏻

Reply:
Tracked with [IMN-184](https://pagopa.atlassian.net/browse/IMN-184)